Db/test (#1)
dutterbutter/zksync-web-era-docs: docs/dev/tutorials/api3-usd-paymaster-tutorial.md

:::

## Project repo

The tutorial code is available [here](https://github.com/vanshwassan/zk-paymaster-dapi-poc)

## Set up the project

1. We're going to use [zkSync CLI](/docs/tools/zksync-cli/) to set up an empty project. Install it globally:

```sh
$ yarn add global zksync-cli@latest
```

2. After installation, run the following command to create a new project:

```sh
$ yarn zksync-cli create paymaster-dapi
```

3. This will create a new zkSync project called `paymaster-dapi` with a basic `Greeter` contract. `cd` into the project directory:

```sh
$ cd paymaster-dapi
```

3. Add the project dependencies, including openzeppelin, zkSync and API3 contracts:

```sh
$ yarn add -D @matterlabs/zksync-contracts @openzeppelin/contracts @openzeppelin/contracts-upgradeable @api3/contracts dotenv
```

## Design

For the sake of simplicity, we will use a modified OpenZeppelin ERC20 implementation. For that, we are going to code a basic ERC20 token `mockUSDC` which will be used to pay for the transactions.

> 1. Create a new contract `mockUSDC.sol` under `/contracts` directory and add the following code:

Author: ```suggestion
1. Within the `/contracts` directory, create a new file named `mockUSDC.sol`:
```
cd contracts/ && touch mockUSDC.sol
```
```
